src/main.cpp

```diff
@@ -91,6 +61,23 @@ int main(int argc, char **argv){
         }
     }
 
+    // Create jointstate publisher (for use with state_publisher node)
+    ros::NodeHandle nodehandle;
+    ros::Publisher JointPublisher = nodehandle.advertise<sensor_msgs::JointState>("joint_states", 10, false);
+    ros::Publisher marker_pub = nodehandle.advertise<visualization_msgs::MarkerArray>("geometry", 10);
+
+    // Create location broadcaster
+    tf2_ros::TransformBroadcaster LocationBroadcaster;
+
+    double mapOffsetX, mapOffsetY, mapRotation;
+    {
+        cv::Mat image = cv::imread(heightmap_filename, cv::IMREAD_GRAYSCALE);
+        mapOffsetX = (image.cols * 0.025 / 2);
+        mapOffsetY = (image.rows * 0.025 / 2);
+        mapRotation = 1.570796;
+        
+    }
```

Comment:
Get this info from the `map_metadata` topic http://wiki.ros.org/map_server#Published_Topics

Reply:
This is not robust for different map resolution/orientation